secboot/secboot_tpm.go

```diff
@@ -347,24 +347,30 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(
 // UnlockEncryptedVolumeUsingKey unlocks an existing volume using the provided key. The
 // path to the device node is returned.
 // TODO: use UnlockResult here too?
-func UnlockEncryptedVolumeUsingKey(disk disks.Disk, name string, key []byte) (string, error) {
+func UnlockEncryptedVolumeUsingKey(disk disks.Disk, name string, key []byte) (UnlockResult, error) {
 	// find the encrypted device using the disk we were provided - note that
 	// we do not specify IsDecryptedDevice in opts because here we are
 	// looking for the encrypted device to unlock, later on in the boot
 	// process we will look for the decrypted device to ensure it matches
 	// what we expected
 	partUUID, err := disk.FindMatchingPartitionUUID(name + "-enc")
 	if err != nil {
-		return "", err
+		return UnlockResult{}, err
+	}
+	unlockRes := UnlockResult{
+		IsDecryptedDevice: true,
 	}
 	// we have a device
 	encdev := filepath.Join("/dev/disk/by-partuuid", partUUID)
```

Comment:
for consistency with the other Unlock* methods this should also be set on res.Device I think

Reply:
this will be taken care in follow ups